Converted installation scripts to use recipes (discovery).

richardgaunt opened the pull request:
https://www.drupal.org/project/civictheme/issues/3517643

## Checklist before requesting a review

- [ ] I have formatted the subject to include ticket number as `Issue #123456 by drupal_org_username: Issue title`
- [x] I have added a link to the issue tracker
- [ ] I have provided information in `Changed` section about WHY something was done if this was not a normal implementation
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run new and existing relevant tests locally with my changes, and they passed
- [ ] I have provided screenshots, where applicable

## Changed

1. Investigation into recipes and how we could replace the scripts and modules supporting this development environment.
2. Added default content to recipes


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Chores**
  * Updated Docker build configs and ignore rules to include recipes alongside patches and scripts.
  * Updated environment to point the Drupal site setup profile to the new recipe.

* **New Features**
  * Added Civictheme default content, theme settings, and brand color palettes.
  * Added many generated static content items, media assets, and a "Generated content" listing (views) for pages, events, and components.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

richardgaunt committed:
Convert installation scripts to use recipes.

richardgaunt committed:
Add recipes to docker container.

richardgaunt committed:
Updated path to recipes.

richardgaunt committed:
Merge branch 'develop' into feature/convert-site-install-to-recipes

joshua-salsadigital committed:
Merge branch 'develop' into feature/convert-site-install-to-recipes

joshua1234511 committed:
Minor Fixes.

joshua1234511 committed:
Test error [cli stage-0 4/22] COPY recipes /app/recipes:
failed to solve: failed to compute cache key: failed to calculate checksum of ref 2de501c9-428c-4087-bad9-d5742797b5c4::tuluteryoe5mfio86g3g51tdw: "/recipes": not found

joshua1234511 committed:
Fixed recipe installation issues.

joshua1234511 committed:
phpcs fixes

joshua1234511 committed:
Fixed config issues.

joshua1234511 committed:
Fixed node dependency issue in recipie

joshua1234511 committed:
Updated behat test.

joshua1234511 committed:
Fix route null error.

joshua1234511 committed:
Fix route null error.

joshua1234511 committed:
Fix route null error.

richardgaunt committed:
Merge branch 'develop' into feature/convert-site-install-to-recipes

richardgaunt committed:
Merge branch 'develop' into feature/convert-site-install-to-recipes

richardgaunt committed:
Merge branch 'develop' into feature/convert-site-install-to-recipes

joshua-salsadigital committed:
Merge branch 'develop' into feature/convert-site-install-to-recipes

IvanZugec committed:
Added the media_library view for the media field.

IvanZugec commented on recipes/civictheme_site_install/recipe.yml:
```diff
@@ -30,6 +30,8 @@ config:
       - core.entity_form_mode.media.media_library
       - core.entity_view_mode.media.media_library
+      - image.style.media_library
+      - views.view.media_library
```
This fixes a fatal error when you click on the "Add Media" button on media fields.

The media library view wasn't being imported.

joshua-salsadigital committed:
Merge branch 'develop' into feature/convert-site-install-to-recipes

joshua-salsadigital committed:
Merge branch 'develop' into feature/convert-site-install-to-recipes

joshua-salsadigital committed:
Merge branch 'develop' into feature/convert-site-install-to-recipes

joshua-salsadigital committed:
Merge branch 'develop' into feature/convert-site-install-to-recipes

joshua-salsadigital committed:
Merge branch 'develop' into feature/convert-site-install-to-recipes

joshua-salsadigital committed:
Merge branch 'develop' into feature/convert-site-install-to-recipes

richardgaunt committed:
[#3543470] Updated provision callbacks to provide the theme_name to apply provisioning to. (#1416)

richardgaunt committed:
Issue #3543729 by richardgaunt, ivan zugec: Add a script to automate setting up of GovCMS site (#1417)

dependabot[bot] committed:
Bump the npm_and_yarn group across 2 directories with 1 update (#1420)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

anirudhsingh-7773 committed:
Issue #3525185 by anirudhsingh19: Expose banner type as a variable (#1418)

richardgaunt committed:
[#3525687] Fixed cleanup delete cmd (#1421)

richardgaunt committed:
Updated script to be able to skip provisioning content. (#1423)

stooit committed:
Add Quant Cloud migration configuration

- Add quant.type labels to docker-compose services
- Add GitHub Actions build-deploy workflow
- Update Drupal settings for Quant Cloud environment variables
- Make Elasticsearch compatible with ECS Fargate
- Convert post-rollout tasks to entrypoint scripts

stooit committed:
Add Quant Cloud migration configuration

- Add quant.type labels to docker-compose services
- Add GitHub Actions build-deploy workflow
- Update Drupal settings for Quant Cloud environment variables
- Make Elasticsearch compatible with ECS Fargate
- Convert post-rollout tasks to entrypoint scripts

stooit committed:
Add Quant Cloud migration configuration

- Add quant.type labels to docker-compose services
- Add GitHub Actions build-deploy workflow
- Update Drupal settings for Quant Cloud environment variables
- Make Elasticsearch compatible with ECS Fargate
- Convert post-rollout tasks to entrypoint scripts

dependabot[bot] committed:
Bump axios from 1.9.0 to 1.12.0 in /scripts/civictheme-uikit-version-manager in the npm_and_yarn group across 1 directory (#1424)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mlncn committed:
Fix link to theme docs (#1419)

richardgaunt committed:
UIKit bump.

richardgaunt committed:
Updated unit tests for new dev host. (#1426)

alan-cole committed:
Issue #3545860: Added collapsible video player transcript. (#1422)

Co-authored-by: rgaunt <richard.gaunt@protonmail.com>

stooit committed:
Add Quant Cloud migration configuration

- Add quant.type labels to docker-compose services
- Add GitHub Actions build-deploy workflow
- Update Drupal settings for Quant Cloud environment variables
- Make Elasticsearch compatible with ECS Fargate
- Convert post-rollout tasks to entrypoint scripts
- Add nginx configuration for CDN header handling
- Add Quant Cloud scripts (provision and database backup)

stooit committed:
Add Quant Cloud migration configuration

- Add quant.type labels to docker-compose services
- Add GitHub Actions build-deploy workflow
- Update Drupal settings for Quant Cloud environment variables
- Make Elasticsearch compatible with ECS Fargate
- Convert post-rollout tasks to entrypoint scripts
- Add nginx configuration for CDN header handling
- Add Quant Cloud scripts (provision and database backup)

richardgaunt committed:
Updated content.

richardgaunt committed:
Added recipe for generated static content.

richardgaunt committed:
Merge branch 'feature/generated-content-static' into feature/convert-site-install-to-recipes

richardgaunt committed:
Added missing view, updated message test.

richardgaunt committed:
Updated static content.

richardgaunt committed:
Updated for faulty static content.

richardgaunt committed:
Merge branch 'feature/generated-content-static' into feature/convert-site-install-to-recipes

richardgaunt committed:
Added default content to content provision.

richardgaunt committed:
Removed incorrect provisioning of default content.

richardgaunt committed:
Fixed test, added content view to site install.

richardgaunt committed:
Added config action to recipes.

richardgaunt committed:
Updates to assets for content recipes.

richardgaunt committed:
Merge branch 'develop' into feature/convert-site-install-to-recipes

richardgaunt committed:
Update text_format.

richardgaunt committed:
Fixed failing tests.

richardgaunt committed:
Updated generated content recipe.

richardgaunt committed:
Remove content recipes.

richardgaunt committed:
Revert unnecessary changes.

richardgaunt committed:
Add back search module to civictheme.

richardgaunt committed:
Added back content recipes.

richardgaunt committed:
Removing default content module.

richardgaunt committed:
Added additional config for recipe, updated xss test.

richardgaunt committed:
Merge branch 'develop' into feature/convert-site-install-to-recipes
